paddle/contrib/inference/paddle_inference_api.h

```diff
@@ -12,49 +12,58 @@
    See the License for the specific language governing permissions and
    limitations under the License. */
 
+/*
+ * This file contains the definition of a simple Inference API for Paddle.
+ *
+ * ATTENTION: It requires some C++ features, for lower version C++ or C, we
+ * might release another API.
+ */
+
 #pragma once
 
+#include <memory>
 #include <string>
 #include <vector>
 
-namespace paddle {
+struct PaddleTensor {
+  std::string name;  // variable name.
+  std::vector<int> shape;
+  std::vector<unsigned char> data;         // bytes of data.
+  size_t type{typeid(float).hash_code()};  // hash of type
```

Comment:
not sure typeid is a good idea. What are the typeid for float16 and int8, and mabe complex32 in the future...

Maybe enum?

Reply:
Yes, the `typeid` is unique for a different type, it is a global id compared to an enum.

 > size_t type{typeid(float).hash_code()};

this is a default value, users can set with a `typeid` of any type.